### PR TITLE
ATARI environments - part1

### PR DESCRIPTION
**Pull request by JulienT01**
PART1.
Add and update code to train on atari games (DQN and PPO on ALE/Breakout and ALE/Freewy):
- Udate 'agents/torch/utils/models.py' to have better management of the size to connect the CNN and the head (MLP). And to manage the "multi-batch" dimensions. Atari wrapper give input by batch (for the dynamics), and DQN use chunk, so we have to merge this 2 kind of batch in 1, make the forward, then split the result in the previous batches
- update 'agents/torch/utils/training.py' to manage the 'transpose_obs' parameters in automatic model_config (and don't overwritte previous settings)

TODO part 2 (Other PR) :  https://github.com/rlberry-py/rlberry/pull/285
- update atari_make (in gym_make.py) and 'DQN agent' to generate and manage vectorized environments.
- update A2C and PPO agent to manage the vectorized environments

### CODE REVIEW

File: rlberry/agents/torch/utils/models.py [hunk 1]
```diff
@@ -434,26 +436,32 @@ def __init__(
...
+        """
+        conv_result = self.activation((self.conv1(torch.zeros(1, *shape))))
+        conv_result = self.activation((self.conv2(conv_result)))
+        conv_result = self.activation((self.conv3(conv_result)))
+        return int(np.prod(conv_result.size()))
+
     def convolutions(self, x):
         x = x.float()
-        if len(x.shape) == 3:
+        if (
```
Comment (mmcenta) [inline rlberry/agents/torch/utils/models.py:462]: maybe move this comment to the line above so black doesn't split this into three lines

File: rlberry/envs/gym_make.py [hunk 2]
```diff
@@ -32,14 +33,59 @@ def gym_make(id, wrap_spaces=False, **kwargs):
...
+    # if scalarize is None:
+    #     if "n_envs" in kwargs.keys() and int(kwargs["n_envs"])>1:
+    #         scalarize = False
+    #     else:
+    #         scalarize = True
+
+    scalarize = True
+
+    if "atari_wrappers_dict" in kwargs.keys():
+        atari_wrappers_dict = kwargs["atari_wrappers_dict"]
```
Comment (mmcenta) [inline rlberry/envs/gym_make.py:50]: You can just do `atari_wrappers_dict = kwargs.pop('atari_wrappers_dict')` for the same effect.

File: rlberry/envs/gym_make.py [hunk 3]
```diff
@@ -32,14 +33,59 @@ def gym_make(id, wrap_spaces=False, **kwargs):
     return Wrapper(env, wrap_spaces=wrap_spaces)
 
 
-def atari_make(id, scalarize=True, **kwargs):
+def atari_make(id, scalarize=None, **kwargs):
```
Comment (mmcenta) [inline rlberry/envs/gym_make.py:36]: Why `True` to `None`? Shouldn't it be `False`?

**Review by mmcenta — APPROVED** [review]
Just one note: I need help understanding what the ScalarizeEnvWrapper is doing here. It feels like it really only should be used when `n_envs = 1`; otherwise, you're giving the same action to potentially different environments, right?

Otherwise, LGTM.

File: rlberry/envs/gym_make.py [hunk 4]
```diff
@@ -32,14 +33,58 @@ def gym_make(id, wrap_spaces=False, **kwargs):
     return Wrapper(env, wrap_spaces=wrap_spaces)
 
 
-def atari_make(id, scalarize=True, **kwargs):
+def atari_make(id, scalarize=False, **kwargs):
```
Comment (TimotheeMathieu) [inline rlberry/envs/gym_make.py:36]: A small docstring?

**Review by TimotheeMathieu — APPROVED** [review]
A small comment, otherwise LGTM.

**Pull request merged**